aws_advanced_python_wrapper/host_monitoring_v2_plugin.py

```diff
@@ -0,0 +1,506 @@
+        self._driver_dialect: DriverDialect = self._plugin_service.driver_dialect
+
+        self._monitor_run_thread: Thread = Thread(daemon=True, name="HostMonitoringThreadRun", target=self.run)
+        self._monitor_run_thread.start()
+        self._monitor_new_context_thread: Thread = Thread(daemon=True, name="HostMonitoringThreadNewContextRun",
+                                                          target=self._new_context_run)
+        self._monitor_new_context_thread.start()
+
+    def can_dispose(self) -> bool:
+        return self._active_contexts.empty() and len(self._new_contexts.items()) == 0
+
+    @property
+    def is_stopped(self):
+        return self._is_stopped.get()
+
+    def stop(self):
+        self._is_stopped.set(True)
+
+    def start_monitoring(self, context: MonitoringContext):
+        if self.is_stopped:
+            logger.warning("HostMonitorV2.MonitorIsStopped", self._host_info.host)
+
+        current_time_ns = perf_counter_ns()
+        start_monitoring_time_ns = ((current_time_ns + self._failure_detection_time_ns) // 10**9) * 10**9
+        queue: Queue = Queue()
+        self._new_contexts.compute_if_absent(start_monitoring_time_ns, lambda k: queue)
+        queue.put(weakref.ref(context))
+
+    def get_current_time_ns(self) -> float:
+        return float(perf_counter_ns())
+
+    def _new_context_run(self) -> None:
+        logger.debug("HostMonitorV2.StartMonitoringThreadNewContext", self._host_info.host)
+
+        try:
+            while not self.is_stopped:
+                current_time_ns = self.get_current_time_ns()
+
+                processed_keys = []
+                keys = list(self._new_contexts.keys())
+                for key in keys:
+                    queue: Queue | None = self._new_contexts.get(key)
+                    if queue is not None:
+                        if key < current_time_ns:
+                            processed_keys.append(key)
+                            while not queue.empty():
+                                context_weak_ref = queue.get()
+                                if context_weak_ref is not None and context_weak_ref() is not None and context_weak_ref().is_active():
+                                    self._active_contexts.put(context_weak_ref)
+
+                for key in processed_keys:
+                    self._new_contexts.remove(key)
+
+                sleep(1)
+        except InterruptedError:
+            pass
+        except Exception:
+            logger.debug("HostMonitorV2.ExceptionDuringMonitoringStop", self._host_info.host)
+
+        logger.debug("HostMonitorV2.StopMonitoringThreadNewContext", self._host_info.host)
+
+    def run(self) -> None:
+        logger.debug("HostMonitorV2.StartMonitoringThread", self._host_info.host)
+
+        try:
+            while not self.is_stopped:
+                if self._active_contexts.empty() and not self._is_unhealthy:
+                    sleep(self._THREAD_SLEEP_NANO // 1_000_000_000)
```

Comment:
`//` (floor division) returns an integer rounded down, since THREAD_SLEEP_NANO is 100ms the result of the division will be 0. We should use `/` (true division) instead here to return a float
```suggestion
                    sleep(self._THREAD_SLEEP_NANO / 1_000_000_000)
```